### ISSUE TITLE
tough-cookie  4.0.0 integration broken

### ISSUE THREAD

**Issue by sneusse** [sindresorhus/got#1131]
Since this commit in tough-cookie (https://github.com/salesforce/tough-cookie/commit/d5d6f793129250858fefb374c4b0905b498c2ff7#diff-d721bcd439ec1bdd00213161a426ebd8 , code below) the following line to check the cookieJar won't work anymore:

https://github.com/sindresorhus/got/blob/62b338848a20390d9f6fba7fee9ee4413a504a79/source/normalize-arguments.ts#L171

as the code in tough-cookie patches the prototype so the length of all the methods will always return 0.

https://github.com/salesforce/tough-cookie/blob/2524513d49b7fab37639dfb7c6b87994c2bd7791/lib/cookie.js#L1618-L1629

The workaround provided here #1082 will also work here. 

-- one more workaround:

```javascript
function proxy(org, proxifier) {
  return new Proxy(proxifier(org), {
    get: (obj, prop) => (prop in obj ? obj[prop] : org[prop])
  });
}

const cookies = proxy(new CookieJar(), obj => {
  return {
    setCookie: async (rawCookie, url) => obj.setCookie(rawCookie, url, console.log),
    getCookieString: async url => obj.getCookieString(url)
  };
});

const resp = await got("https://google.com", { cookieJar: cookies });

```

**Comment by szmarczak** [comment]
Fixed in 407d5972255bdc052ff2b081b12a5fea32f908b2

**Comment by szmarczak** [comment]
Fixed in 9eb8407787eac3026a9c27e44668716d12fceb71
